sqlmesh/core/plan/builder.py

```diff
@@ -0,0 +1,617 @@
+        environment_ttl: The period of time that a development environment should exist before being deleted.
+        categorizer_config: Auto categorization settings.
+        auto_categorization_enabled: Whether to apply auto categorization.
+        effective_from: The effective date from which to apply forward-only changes on production.
+        include_unmodified: Indicates whether to include unmodified nodes in the target development environment.
+        environment_suffix_target: Indicates whether to append the environment name to the schema or table name.
+        default_start: The default plan start to use if not specified.
+        default_end: The default plan end to use if not specified.
+    """
+
+    def __init__(
+        self,
+        context_diff: ContextDiff,
+        start: t.Optional[TimeLike] = None,
+        end: t.Optional[TimeLike] = None,
+        execution_time: t.Optional[TimeLike] = None,
+        apply: t.Optional[t.Callable[[Plan], None]] = None,
+        restate_models: t.Optional[t.Iterable[str]] = None,
+        backfill_models: t.Optional[t.Iterable[str]] = None,
+        no_gaps: bool = False,
+        skip_backfill: bool = False,
+        is_dev: bool = False,
+        forward_only: bool = False,
+        environment_ttl: t.Optional[str] = None,
+        environment_suffix_target: EnvironmentSuffixTarget = EnvironmentSuffixTarget.default,
+        environment_catalog_mapping: t.Optional[t.Dict[re.Pattern, str]] = None,
+        categorizer_config: t.Optional[CategorizerConfig] = None,
+        auto_categorization_enabled: bool = True,
+        effective_from: t.Optional[TimeLike] = None,
+        include_unmodified: bool = False,
+        default_start: t.Optional[TimeLike] = None,
+        default_end: t.Optional[TimeLike] = None,
+    ):
+        self._context_diff = context_diff
+        self._no_gaps = no_gaps
+        self._skip_backfill = skip_backfill
+        self._is_dev = is_dev
+        self._forward_only = forward_only
+        self._environment_ttl = environment_ttl
+        self._categorizer_config = categorizer_config or CategorizerConfig()
+        self._auto_categorization_enabled = auto_categorization_enabled
+        self._include_unmodified = include_unmodified
+        self._restate_models = set(restate_models or [])
+        self._effective_from = effective_from
+        self._execution_time = execution_time
+        self._backfill_models = backfill_models
+        self._end = end or default_end
+        self._apply = apply
+
+        self._start = start
+        if not self._start and is_dev and forward_only:
+            self._start = default_start or yesterday_ds()
+
+        self._environment_naming_info = EnvironmentNamingInfo.from_environment_catalog_mapping(
+            environment_catalog_mapping or {},
+            name=self._context_diff.environment,
+            suffix_target=environment_suffix_target,
+        )
+
+        self._override_start = start is not None
```

Comment:
just make these public and delete the properties

Reply:
This is intentional to signal that these values are not supposed to be modified

Reply:
meh, no one's gonna modify it

Reply:
famous last words lol. Ok will change this

Reply:
Interesting 👀